src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+function prepareItems(
+  allItems: string[],
+  itemsPerPage: number,
+  currentPage: number,
+) {
+  const start = itemsPerPage * currentPage - itemsPerPage;
+  const end =
+    start + itemsPerPage > allItems.length
+      ? allItems.length
+      : start + itemsPerPage;
+
+  return allItems.slice(start, end);
+}
+
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const visibleItems = prepareItems(items, itemsPerPage, currentPage);
+
+  const startItem = itemsPerPage * currentPage - itemsPerPage;
+  const endItem =
+    startItem + itemsPerPage > items.length
+      ? items.length
+      : startItem + itemsPerPage;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startItem + 1} - {endItem} of {items.length})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={event => {
```

Comment:
When changing the number of items per page, you are directly setting the state to the numeric value of the event target's value. It's good practice to add some validation to ensure that the value is indeed a number and within an expected range to prevent potential bugs if the select options were to change in the future.